engine/access/state_stream/backend/handler.go

```diff
@@ -213,3 +215,155 @@ func (h *Handler) GetRegisterValues(_ context.Context, request *executiondata.Ge
+		}
+
+		results = append(results, &executiondata.SubscribeAccountStatusesResponse_Result{
+			Address: flow.HexToAddress(address).Bytes(),
+			Events:  convertedEvent,
+		})
+	}
+	return results, nil
+}
+
+// sendSubscribeAccountStatusesResponseFunc defines the function signature for sending account status responses
+type sendSubscribeAccountStatusesResponseFunc func(*executiondata.SubscribeAccountStatusesResponse) error
+
+// handleAccountStatusesResponse handles account status responses by converting them to the message and sending them to the subscriber.
+func (h *Handler) handleAccountStatusesResponse(
+	requestHeartbeatInterval uint64,
+	evenVersion entities.EventEncodingVersion,
+	send sendSubscribeAccountStatusesResponseFunc,
+) func(resp *AccountStatusesResponse) error {
+	heartbeatInterval := requestHeartbeatInterval
+	if heartbeatInterval == 0 {
+		heartbeatInterval = h.defaultHeartbeatInterval
+	}
+
+	blocksSinceLastMessage := uint64(0)
+
+	return func(resp *AccountStatusesResponse) error {
+		// check if there are any events in the response. if not, do not send a message unless the last
+		// response was more than HeartbeatInterval blocks ago
+		if len(resp.AccountEvents) == 0 {
+			blocksSinceLastMessage++
+			if blocksSinceLastMessage < heartbeatInterval {
+				return nil
+			}
+			blocksSinceLastMessage = 0
+		}
+
+		results, err := convertAccountsStatusesResultsToMessage(evenVersion, resp)
+		if err != nil {
+			return err
+		}
+
+		err = send(&executiondata.SubscribeAccountStatusesResponse{
+			BlockId:      convert.IdentifierToMessage(resp.BlockID),
+			BlockHeight:  resp.Height,
+			Results:      results,
+			MessageIndex: resp.MessageIndex,
+		})
+		if err != nil {
+			return rpc.ConvertError(err, "could not send response", codes.Internal)
+		}
+
+		return nil
+	}
+}
+
+// SubscribeAccountStatusesFromStartBlockID streams account statuses for all blocks starting at the requested
+// start block ID, up until the latest available block. Once the latest is
+// reached, the stream will remain open and responses are sent for each new
+// block as it becomes available.
+func (h *Handler) SubscribeAccountStatusesFromStartBlockID(
+	request *executiondata.SubscribeAccountStatusesFromStartBlockIDRequest,
+	stream executiondata.ExecutionDataAPI_SubscribeAccountStatusesFromStartBlockIDServer,
+) error {
+	// check if the maximum number of streams is reached
+	if h.StreamCount.Load() >= h.MaxStreams {
+		return status.Errorf(codes.ResourceExhausted, "maximum number of streams reached")
+	}
+
+	h.StreamCount.Add(1)
+	defer h.StreamCount.Add(-1)
+
+	startBlockID, err := convert.BlockID(request.GetStartBlockId())
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "could not convert start block ID: %v", err)
+	}
+
+	statusFilter := request.GetFilter()
+	filter, err := state_stream.NewAccountStatusFilter(h.eventFilterConfig, h.chain, statusFilter.GetEventType(), statusFilter.GetAddress())
```

Comment:
In https://github.com/onflow/flow-go/pull/5406/files#diff-982b9e2e612badc329395c0f96312d29aa5ad6f4c1c5427327206d3706769e00R38, you have defined a helper function to reduce boilerplate when subscribing is it possible to use similar approach here?